File: angular/src/app/book.ts
export class Book {
    
    id!: number;
    name: string;
    authorName: string;
    publicationDate: string;
    copies: number;

    constructor() {
        this.name="";
        this.authorName="";
        this.publicationDate="";
        this.copies=0;
    }
}

File: backend/src/main/java/edu/baylor/cs/se/hibernate/model/Book.java
package edu.baylor.cs.se.hibernate.model;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import lombok.Data;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import java.time.LocalDate;

@Data
@Entity
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Book {
    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String authorName;

    @Column(nullable = false)
    private LocalDate publicationDate;

    @Column(nullable = false)
    private Integer copies;
}
